### ISSUE TITLE
Docker deployment broken?

### ISSUE THREAD

**Issue by mwarning** [engelsystem/engelsystem#750]
I tried different docker deployment approaches without much success:

```
git clone https://github.com/engelsystem/engelsystem.git
cd engelsystem
cd docker
docker-compose build
```
At this pint, accessing http://localhost:8080 will yield an error message: "An <del>un</del>expected error occurred. A team of untrained monkeys has been dispatched to fix it."

The https://github.com/engelsystem/engelsystem/blob/master/DEVELOPMENT.md instructions for Docker get me to the web login, but this is labeled and meant for development...

The  v3.1.0 release seems to miss some files, `docker-compose build` does not finish:
```
 ---> a24bb4013296
Step 10/31 : COPY .babelrc .browserslistrc composer.json LICENSE package.json README.md webpack.config.js yarn.lock /app/
ERROR: Service 'es_php_fpm' failed to build: COPY failed: stat /var/lib/docker/tmp/docker-builder441887043/.babelrc: no such file or directory
```
(Also missing are yarn.lock and webpack.config.js)

**Comment by mwarning** [comment]
After I get the missing files from master, I do the following:
```
docker network create --driver bridge heaven
docker run --network heaven --name es_php_fpm es_php_fpm
docker run --network heaven -p 8080:80 --name es_nginx es_nginx
```

When I now go to http://localhost:8080 now, I get  "An unexpected error occurred. A team of untrained monkeys has been dispatched to fix it.".

**Comment by mwarning** [comment]
The error message that appears on the console:

```
docker run --network heaven --name es_php_fpm es_php_fpm
[04-Aug-2020 14:54:39] NOTICE: fpm is running, pid 1
[04-Aug-2020 14:54:39] NOTICE: ready to handle connections
NOTICE: PHP message: Exception: Code: 0, Message: Error: Unable to connect to database, File: src/Database/DatabaseServiceProvider.php:63, Trace: [{"file":"\/var\/www\/src\/Database\/DatabaseServiceProvider.php","line":41,"function":"exitOnError","class":"Engelsystem\\Database\\DatabaseServiceProvider","type":"->","args":[]},{"file":"\/var\/www\/src\/Application.php","line":68,"function":"register","class":"Engelsystem\\Database\\DatabaseServiceProvider","type":"->","args":[]},{"file":"\/var\/www\/src\/Application.php","line":90,"function":"register","class":"Engelsystem\\Application","type":"->","args":[{}]},{"file":"\/var\/www\/includes\/application.php","line":21,"function":"bootstrap","class":"Engelsystem\\Application","type":"->","args":[{"providers":["Engelsystem\\Logger\\LoggerServiceProvider","Engelsystem\\Exceptions\\ExceptionsServiceProvider","Engelsystem\\Config\\ConfigServiceProvider","Engelsystem\\Http\\UrlGeneratorServiceProvider","Engelsystem\\Renderer\\RendererServiceProvider","Engelsystem\\Database\\DatabaseServiceProvider","Engelsystem\\Http\\RequestServiceProvider","Engelsystem\\Http\\SessionServiceProvider","Engelsystem\\Helpers\\Translation\\TranslationServiceProvider","Engelsystem\\Http\\ResponseServiceProvider","Engelsystem\\Http\\Psr7ServiceProvider","Engelsystem\\Helpers\\AuthenticatorServiceProvider","Engelsystem\\Renderer\\TwigServiceProvider","Engelsystem\\Middleware\\RouteDispatcherServiceProvider","Engelsystem\\Middleware\\RequestHandlerServiceProvider","Engelsystem\\Middleware\\SessionHandlerServiceProvider","Engelsystem\\Http\\Validation\\ValidationServiceProvider","Engelsystem\\Http\\RedirectServiceProvider","Engelsystem\\Helpers\\VersionServiceProvider","Engelsystem\\Mail\\MailerServiceProvider","Engelsystem\\Http\\HttpClientServiceProvider"],"middleware":["Engelsystem\\Middleware\\SendResponseHandler","Engelsystem\\Middleware\\ExceptionHandler","Engelsystem\\Middleware\\SetLocale","Engelsystem\\Middleware\\AddHeaders","Engelsystem\\Middleware\\ErrorHandler","Engelsystem\\Middleware\\VerifyCsrfToken","Engelsystem\\Middleware\\RouteDispatcher","Engelsystem\\Middleware\\SessionHandler","Engelsystem\\Middleware\\RequestHandler"]}]},{"file":"\/var\/www\/includes\/engelsystem.php","line":6,"args":["\/var\/www\/includes\/application.php"],"function":"require"},{"file":"\/var\/www\/public\/index.php","line":7,"args":["\/var\/www\/includes\/engelsystem.php"],"function":"require_once"}]
172.18.0.3 -  04/Aug/2020:14:55:00 +0000 "GET /index.php" 500
NOTICE: PHP message: Exception: Code: 0, Message: Error: Unable to connect to database, File: src/Database/DatabaseServiceProvider.php:63, Trace: [{"file":"\/var\/www\/src\/Database\/DatabaseServiceProvider.php","line":41,"function":"exitOnError","class":"Engelsystem\\Database\\DatabaseServiceProvider","type":"->","args":[]},{"file":"\/var\/www\/src\/Application.php","line":68,"function":"register","class":"Engelsystem\\Database\\DatabaseServiceProvider","type":"->","args":[]},{"file":"\/var\/www\/src\/Application.php","line":90,"function":"register","class":"Engelsystem\\Application","type":"->","args":[{}]},{"file":"\/var\/www\/includes\/application.php","line":21,"function":"bootstrap","class":"Engelsystem\\Application","type":"->","args":[{"providers":["Engelsystem\\Logger\\LoggerServiceProvider","Engelsystem\\Exceptions\\ExceptionsServiceProvider","Engelsystem\\Config\\ConfigServiceProvider","Engelsystem\\Http\\UrlGeneratorServiceProvider","Engelsystem\\Renderer\\RendererServiceProvider","Engelsystem\\Database\\DatabaseServiceProvider","Engelsystem\\Http\\RequestServiceProvider","Engelsystem\\Http\\SessionServiceProvider","Engelsystem\\Helpers\\Translation\\TranslationServiceProvider","Engelsystem\\Http\\ResponseServiceProvider","Engelsystem\\Http\\Psr7ServiceProvider","Engelsystem\\Helpers\\AuthenticatorServiceProvider","Engelsystem\\Renderer\\TwigServiceProvider","Engelsystem\\Middleware\\RouteDispatcherServiceProvider","Engelsystem\\Middleware\\RequestHandlerServiceProvider","Engelsystem\\Middleware\\SessionHandlerServiceProvider","Engelsystem\\Http\\Validation\\ValidationServiceProvider","Engelsystem\\Http\\RedirectServiceProvider","Engelsystem\\Helpers\\VersionServiceProvider","Engelsystem\\Mail\\MailerServiceProvider","Engelsystem\\Http\\HttpClientServiceProvider"],"middleware":["Engelsystem\\Middleware\\SendResponseHandler","Engelsystem\\Middleware\\ExceptionHandler","Engelsystem\\Middleware\\SetLocale","Engelsystem\\Middleware\\AddHeaders","Engelsystem\\Middleware\\ErrorHandler","Engelsystem\\Middleware\\VerifyCsrfToken","Engelsystem\\Middleware\\RouteDispatcher","Engelsystem\\Middleware\\SessionHandler","Engelsystem\\Middleware\\RequestHandler"]}]},{"file":"\/var\/www\/includes\/engelsystem.php","line":6,"args":["\/var\/www\/includes\/application.php"],"function":"require"},{"file":"\/var\/www\/public\/index.php","line":7,"args":["\/var\/www\/includes\/engelsystem.php"],"function":"require_once"}]
172.18.0.3 -  04/Aug/2020:14:55:01 +0000 "GET /index.php" 500
```

**Comment by mwarning** [comment]
Maybe I had to initialize the database..:
```
docker exec -it es_php_fpm bin/migrate
Exception: Code: 0, Message: Error: Unable to connect to database, File: src/Database/DatabaseServiceProvider.php:63, Trace: [{"file":"\/var\/www\/src\/Database\/DatabaseServiceProvider.php","line":41,"function":"exitOnError","class":"Engelsystem\\Database\\DatabaseServiceProvider","type":"->","args":[]},{"file":"\/var\/www\/src\/Application.php","line":68,"function":"register","class":"Engelsystem\\Database\\DatabaseServiceProvider","type":"->","args":[]},{"file":"\/var\/www\/src\/Application.php","line":90,"function":"register","class":"Engelsystem\\Application","type":"->","args":[{}]},{"file":"\/var\/www\/includes\/application.php","line":21,"function":"bootstrap","class":"Engelsystem\\Application","type":"->","args":[{"providers":["Engelsystem\\Logger\\LoggerServiceProvider","Engelsystem\\Exceptions\\ExceptionsServiceProvider","Engelsystem\\Config\\ConfigServiceProvider","Engelsystem\\Http\\UrlGeneratorServiceProvider","Engelsystem\\Renderer\\RendererServiceProvider","Engelsystem\\Database\\DatabaseServiceProvider","Engelsystem\\Http\\RequestServiceProvider","Engelsystem\\Http\\SessionServiceProvider","Engelsystem\\Helpers\\Translation\\TranslationServiceProvider","Engelsystem\\Http\\ResponseServiceProvider","Engelsystem\\Http\\Psr7ServiceProvider","Engelsystem\\Helpers\\AuthenticatorServiceProvider","Engelsystem\\Renderer\\TwigServiceProvider","Engelsystem\\Middleware\\RouteDispatcherServiceProvider","Engelsystem\\Middleware\\RequestHandlerServiceProvider","Engelsystem\\Middleware\\SessionHandlerServiceProvider","Engelsystem\\Http\\Validation\\ValidationServiceProvider","Engelsystem\\Http\\RedirectServiceProvider","Engelsystem\\Helpers\\VersionServiceProvider","Engelsystem\\Mail\\MailerServiceProvider","Engelsystem\\Http\\HttpClientServiceProvider"],"middleware":["Engelsystem\\Middleware\\SendResponseHandler","Engelsystem\\Middleware\\ExceptionHandler","Engelsystem\\Middleware\\SetLocale","Engelsystem\\Middleware\\AddHeaders","Engelsystem\\Middleware\\ErrorHandler","Engelsystem\\Middleware\\VerifyCsrfToken","Engelsystem\\Middleware\\RouteDispatcher","Engelsystem\\Middleware\\SessionHandler","Engelsystem\\Middleware\\RequestHandler"]}]},{"file":"\/var\/www\/bin\/migrate","line":9,"args":["\/var\/www\/includes\/application.php"],"function":"require_once"}]
An <del>un</del>expected error occurred. A team of untrained monkeys has been dispatched to fix it.
```

**Comment by MyIgel** [comment]
Hi,
after running
```
git clone https://github.com/engelsystem/engelsystem.git
cd engelsystem/docker
docker-compose build
docker-compose up -d
docker exec -it engelsystem_es_php_fpm_1 bin/migrate
```
as stated in the readme (the only missing part is the `docker-compose up -d` here) i got a working Engelsystem

Besides that: The other files are in fact not included in the release zip as its the "minimum deployable Engelsystem" and thus without the files, the docker directory was more or less a leftover ;)

Tldr: The master branch can be run with docker-compose and  works, the release zip is more for "directly deploying it to a web server"

As you started the containers manually you have to link them together which seems to be the problem here?

**Comment by mwarning** [comment]
```
git clone https://github.com/engelsystem/engelsystem.git
cd engelsystem/docker
docker-compose build
docker-compose up -d
docker exec -it engelsystem_es_php_fpm_1 bin/migrate
```
this works - maybe update the Readme? ;-)

thanks

**Comment by MyIgel** [comment]
Jup, i will add the `docker-compose up -d` part to it later :p

**Comment by mwarning** [comment]
Thanks. Also change  "Import database" to e.g. "Now import the database", to express this is a necessary step.
